abci/multiplexer.go

```diff
@@ -136,54 +405,34 @@ func (m *Multiplexer) getApp() (servertypes.ABCI, error) {
 	// get the appropriate version for the latest app version.
 	currentVersion, err := m.versions.GetForAppVersion(m.lastAppVersion)
 	if err != nil {
-		m.logger.Info("No app found in multiplexer for app version; using latest app", "app_version", m.lastAppVersion, "err", err)
-		panic("TOOD: start latest app here")
-		return m.latestApp, nil
-	}
-
-	// check if we need to start the app or if we have a different app running
-	if !m.started || currentVersion.AppVersion != m.activeVersion.AppVersion {
-		if currentVersion.Appd == nil {
-			return nil, fmt.Errorf("appd is nil for version %d", m.activeVersion.AppVersion)
-		}
-
-		// check if an app is already started
-		// stop the app if it's running
-		if m.activeVersion.Appd != nil && m.activeVersion.Appd.Pid() != appd.AppdStopped {
-			m.logger.Info("Stopping app for version", "maximum_app_version", m.activeVersion.AppVersion)
-			if err := m.activeVersion.Appd.Stop(); err != nil {
-				return nil, fmt.Errorf("failed to stop app for version %d: %w", m.activeVersion.AppVersion, err)
-			}
-			m.started = false
-			m.activeVersion = Version{}
+		// if we are switching from an embedded binary to a native one, we need to ensure that we stop it
+		// before we start the native app.
+		if err := m.stopActiveVersion(); err != nil {
```

Comment:
this is the core change, we stop any embedded version,  e.g. v3 and then we start a native v4 and start up the grpc/api servers.